src/pages/AdminBooksPage/BooksTable/index.tsx

```diff
@@ -1,43 +1,108 @@
+import { ChangeEvent, useRef, useState } from "react";
 import BooksTableHead from "./BooksTableHead";
 import BooksTableBody from "./BooksTableBody";
+import useBooks from "./useBooks";
+import BooksTableSkeleton from "./BooksTableSkeleton/BooksTableSkeleton";
+import ConfirmationModal, {
+  ActionVariant,
+  ConfirmationRefObject,
+} from "../../../components/Modals/ConfirmationModal";
+import { warningMessages } from "../../../constants/warningMessages";
+import { ConfirmationMessages } from "../../../constants/confirmationMessages";
 
 const BooksTable = () => {
-  const { books, allBooksCount, page, setPage } = useBooksContext();
+  const [page, setPage] = useState<number>(0);
+  const [selectedId, setSelectedId] = useState<number>();
+
   const [rowsPerPage, setRowsPerPage] = useState<number>(18);
   const rowsOptions = [18];
   const columns = ["Image", "Name", "Author", "Status", "Rating", ""];
 
+  const modal = useRef<ConfirmationRefObject>(null);
+  const showModal = () => modal?.current?.showModal?.();
+  const hideModal = () => modal?.current?.hideModal?.();
+
+  const { books, count, loading, deleteBook, findBooks } = useBooks(page);
+
+  const [action, setAction] = useState<ActionVariant>(ActionVariant.delete);
+  const [title, setTitle] = useState<string>("");
+  const [description, setDescription] = useState<string>("");
+
   const handleChangePage = (event: unknown, newPage: number) => {
-    setPage?.(newPage, true);
+    setPage(newPage);
   };
 
   const handleChangeRowsPerPage = (event: ChangeEvent<HTMLInputElement>) => {
     const rows = parseInt(event.target.value, 10);
     setRowsPerPage(rows);
-    setPage?.(0, true);
+    setPage(0);
+  };
+
+  const setConfirmationModal = () => {
+    setAction(ActionVariant.delete);
+    setTitle(ConfirmationMessages.DELETE_BOOK_TITLE);
+    setDescription(ConfirmationMessages.DELETE_BOOK);
+  };
+
+  const setWarningModal = (message: string) => {
+    setAction(ActionVariant.notify);
+    setTitle(warningMessages.DELETE_BOOK_TITLE);
+    setDescription(message);
+  };
+
+  const handleDelete = (selectedId: number) => {
+    setSelectedId(selectedId);
+    setConfirmationModal();
+    showModal();
   };
 
+  const handleConfirm = () => {
+    if (selectedId) {
+      deleteBook(selectedId)
+        .then(() => {
+          hideModal();
+          findBooks();
+        })
+        .catch((e) => setWarningModal(e.response.data.message));
+    }
+  };
+
+  const handleEdit = () => {};
+
   return (
     <div>
       <TableContainer component={Paper}>
         <Table aria-label="simple table" data-testid="books-table">
           <BooksTableHead columns={columns} />
-          <BooksTableBody books={books} />
+          {loading ? (
+            <BooksTableSkeleton />
+          ) : (
+            <BooksTableBody
+              books={books}
+              onDelete={handleDelete}
+              onEdit={handleEdit}
+            />
+          )}
```

Comment:
This should be in a separate PR, if we merge it like this, problems with skeleton would require revert of delete book feature